src/Icicle/Common/Base.hs

```diff
@@ -49,17 +49,22 @@ data BaseValue
  | VString T.Text
  | VArray [BaseValue]
  | VPair  BaseValue BaseValue
+ | VLeft  BaseValue
+ | VRight BaseValue
  | VSome  BaseValue
  | VNone
  | VMap    (Map.Map BaseValue    BaseValue)
  | VStruct (Map.Map StructField  BaseValue)
- | VTombstone
 
- | VException ExceptionInfo
+ | VError ExceptionInfo
  deriving (Show, Ord, Eq)
 
+-- | Called "exceptions"
+-- because they aren't really errors,
+-- but they aren't really good values either..
 data ExceptionInfo
```

Comment:
So exceptions in Core aren't errors, but special values? (errors are the `left`s after ReifyPossibilities?)